glean-core/python/glean/testing/__init__.py

```diff
@@ -69,11 +71,23 @@ def __init__(self, file_path):
         self.file_path = file_path
 
     def do_upload(
-        self, url_path: str, serialized_ping: str, configuration: Any
-    ) -> None:
+        self,
+        path: str,
+        data: bytes,
+        headers: List[Tuple[str, str]],
+        config: "Configuration",
+    ) -> ping_uploader.UploadResult:
+        is_gzipped = (
+            len([h for h in headers if h[0] == "Content-Encoding" and h[1] == "gzip"])
+            == 1
+        )
+
+        uncompressed_data = gzip.decompress(data) if is_gzipped else data
         with self.file_path.open("w") as fd:
-            fd.write(str(url_path) + "\n")
-            fd.write(serialized_ping + "\n")
+            fd.write(str(path) + "\n")
+            fd.write(uncompressed_data.decode("utf-8") + "\n")
```

Comment:
Heh, just thinking here but maybe we can avoid writing files here all together. It's for tests only, we could keep stuff in memory (nothing to do here now)

Reply:
In memory won't work since uploading happens in a separate process in the Python bindings...  But like you say, nothing to do here for now.

Reply:
gnah, right, even in tests. Files it is.